cmd/manager/main.go

```diff
@@ -93,18 +93,15 @@ func main() {
 	go http.ListenAndServe(":8080", nil)
 
 	log.Print("Starting the Cmd.")
-	stopCh := signals.SetupSignalHandler()
 
 	go registrySyncer.Sync(stopCh)
 	go catalogsourceconfig.Syncer.Sync(stopCh)
 
-	operatorStatus.SetAvailable(fmt.Sprintf("Version %s of the operator is available", operatorStatus.GetVersion()))
-
 	// Start the Cmd
 	log.Fatal(mgr.Start(stopCh))
 }
 
-func fatal(err error, operatorStatus status.Status) {
-	operatorStatus.SetFailing(err.Error())
+func fatal(err error) {
+	status.SendSyncMessage(err)
```

Comment:
Do we need to do more than send an err to the channel? Don't we still want to set the status to failing somehow?

Reply:
Right now we only send an error on a critical failure, so if the sync ratio is less than one we report failure.

Reply:
I agree with @kevinrizza. A fatal error can happen before `StartReporting()` starts and you would need to set the status to failing.

I am now wondering if we need to do something to be more resilient and retry on initialization failures. But that is for another PR.

Reply:
What the `fatal` function does is write to the channel and then immediately exit. How do we ensure that we give enough time to the other go routine to update the clusteroperator object? Maybe use some sort of `done` channel?

The other question is, we want to update the `clusteroperator` object here, irrespective of sync sync count, right?